lib/evss/disability_compensation_form/data_translation.rb

```diff
@@ -0,0 +1,236 @@
+                         'addressLine1' => address['addressLine1'],
+                         'addressLine2' => address['addressLine2'],
+                         'addressLine3' => address['addressLine3'],
+                         'effectiveDate' => address['effectiveDate'] }
+
+        pciu_address['type'] = get_address_type(address)
+
+        case pciu_address['type']
+        when 'DOMESTIC'
+          zip_code = split_zip_code(address['zipCode'])
+          pciu_address['city'] = address['city']
+          pciu_address['state'] = address['state']
+          pciu_address['zipFirstFive'] = zip_code.first
+          pciu_address['zipLastFour'] = zip_code.last
+        when 'MILITARY'
+          pciu_address['militaryPostOfficeTypeCode'] = address['city']
+          pciu_address['militaryStateCode'] = address['state']
+        when 'INTERNATIONAL'
+          pciu_address['city'] = address['city']
+        end
+
+        pciu_address.compact
+      end
+
+      def get_address_type(address)
+        case address['country']
+        when 'USA'
+          %w[AA AE AP].include?(address['state']) ? 'MILITARY' : 'DOMESTIC'
+        else
+          'INTERNATIONAL'
+        end
+      end
+
+      def split_zip_code(zip_code)
+        zip_code.match(/(^\d{5})(?:([-\s]?)(\d{4})?$)/).captures
+      end
+
+      def convert_treatments
+        form['treatments'].map! do |treatment|
+          treatment['center'] = {
+            'name' => treatment['treatmentCenterName'],
+            'type' => treatment['treatmentCenterType']
+          }
+          treatment['center'].merge!(treatment['treatmentCenterAddress'])
+          treatment['startDate'] = treatment['treatmentDateRange']['from']
+          treatment['endDate'] = treatment['treatmentDateRange']['to']
+          treatment.delete('treatmentCenterName')
+          treatment.delete('treatmentDateRange')
+          treatment.delete('treatmentCenterAddress')
+          treatment.delete('treatmentCenterType')
+          treatment
+        end
+      end
+
+      def application_expiration_date
+        if rad_date.present? && rad_date > application_create_date
+          (rad_date + 1.day + 365.days).iso8601 # expires year after the following day service ends
+        elsif itf.creation_date.nil? || itf.expiration_date.nil? || itf.creation_date > application_create_date
+          (application_create_date + 365.days).iso8601
+        else
+          itf.expiration_date.iso8601
+        end
+      end
+
+      def application_create_date
+        # Application create date is the date the user began their application
+        @acd ||= InProgressForm.where(form_id: VA526ez::FORM_ID, user_uuid: @user.uuid)
+                               .first.created_at
+      end
+
+      def rad_date
+        # retrieve the most recent 'Return from Active Duty' Date
+        @rd ||= Time.zone.parse(@user.military_information.service_episodes_by_date.first&.end_date.to_s)
+      end
+
+      def itf
```

Comment:
What happens if `IntentToFile::Service.get_active` fails? As you mention in the controller the submit process should go into an async job (or multiple chained jobs) but we can't do that until we know that we won't get a validation error from EVSS (i.e. all the validation is on our side)